### PR TITLE
[v2.0.8] [SB] [PD] Ability to manage apps via the Study Builder

### PR DESCRIPTION
**Pull request by monica-BTC**
Added functionality for,

- Study Builder capability to manage multiple apps running off the platform based on App specific properties. Please refer issue #3911
- The ability to configure app-specific email addresses (based on App ID). Please refer issue #3909

### CODE REVIEW

**Comment by aswinijena100** [comment]
@googlebot I consent.

**Comment by madhurya-btc** [comment]
@googlebot I consent.

**Comment by navya-BTC** [comment]
@googlebot I consent.

**Comment by monica-BTC** [comment]
@googlebot I consent.

**Comment by harisboston** [comment]
@googlebot I consent.

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/service/UserServiceImpl.java [hunk 1]
```diff
@@ -447,13 +466,22 @@ private EmailResponse sendAccountLockedEmail(
             ? appConfig.getMailAccountLockedBodyForMobileApp()
             : appConfig.getMailAccountLockedBody();
 
+    String supportEMail =
```
[the remark this answers is not in the document]
Comment (monica-BTC) [reply auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/service/UserServiceImpl.java]: Fixed review comment.

File: participant-manager-datastore/participant-manager-service/src/main/java/com/google/cloud/healthcare/fdamystudies/service/SiteServiceImpl.java [hunk 2]
```diff
@@ -1494,7 +1504,29 @@ public void sendInvitationEmail() {
...
+            resourceLoader
+                .getResource("classpath:Logos/App_Store_Logo.png")
+                .getFile()
+                .getAbsolutePath());
+
+      } catch (IOException e) {
+        logger.error("sendInvitationEmail() failed with an exception.", e);
+      }
+
+      EmailResponse emailResponse = emailService.sendMimeMailWithImages(emailRequest, inlineImages);
```
[the remark this answers is not in the document]
Comment (monica-BTC) [reply participant-manager-datastore/participant-manager-service/src/main/java/com/google/cloud/healthcare/fdamystudies/service/SiteServiceImpl.java]: Fixed review comment.

File: participant-datastore/user-mgmt-module/user-mgmt/src/main/java/com/google/cloud/healthcare/fdamystudies/common/UserMgmntEvent.java [hunk 3]
```diff
@@ -216,7 +216,14 @@
...
       null,
       "Content submitted by app user via 'Contact Us' form in mobile app, could not be emailed to ${contactus_destination_email_address}.",
-      "CONTACT_US_CONTENT_EMAIL_FAILED");
+      "CONTACT_US_CONTENT_EMAIL_FAILED"),
+
+  APP_USER_ACCOUNTS_DELETED(
+      STUDY_BUILDER,
+      PARTICIPANT_USER_DATASTORE,
+      null,
+      "User records for app (with App ID '${AppID}') deactivated on Participant Datastore and deleted from Auth Server, after the app was deactivated via the Study Builder.",
```
[the remark this answers is not in the document]
Comment (monica-BTC) [reply participant-datastore/user-mgmt-module/user-mgmt/src/main/java/com/google/cloud/healthcare/fdamystudies/common/UserMgmntEvent.java]: Fixed review comment.

**Comment by Madhud5** [comment]
@googlebot I consent.

**Comment by Rashi-395** [comment]
@googlebot I consent.

**Review by moschetti — APPROVED** [review]

**Comment by aswinijena100** [comment]
@moschetti Please approve this PR again as resolve conflict stale the earlier approval

**Review by moschetti — APPROVED** [review]

**Pull request merged**